File: targetcalib-pSCT/scripts/mapping/plot_data_tc.py
"""
Plot data using just TargetCalib and python
"""
import numpy as np
from matplotlib import pyplot as plt
from target_calib import CameraConfiguration


def main():

    camera_version = "1.0.1"
    camera_config = CameraConfiguration(camera_version)
    m = camera_config.GetMapping()

    row = np.array(m.GetRowVector())
    col = np.array(m.GetColumnVector())
    nrows = row.max() + 1
    ncols = col.max() + 1

    data = np.loadtxt("input_data.txt")

    image = np.ma.zeros((nrows, ncols))
    image[row, col] = data
    image[0:8, 40:48] = np.ma.masked
    image[0:8, 0:8] = np.ma.masked
    image[40:48, 0:8] = np.ma.masked
    image[40:48, 40:48] = np.ma.masked

    fig = plt.figure(figsize=(10, 10))
    ax = fig.add_subplot(111)
    im = ax.imshow(image, origin='lower')
    fig.colorbar(im)

    axl = m.fOTUpCol_l
    ayl = m.fOTUpRow_l
    adx = m.fOTUpCol_u - axl
    ady = m.fOTUpRow_u - ayl
    ax.arrow(axl, ayl, adx, ady, head_width=1, head_length=1, fc='r', ec='r')
    text = "ON-Telescope UP"
    ax.text(axl, ayl, text, fontsize=8, color='r', ha='center', va='bottom')

    fn = "image_p_plot_data_tc.pdf"
    fig.savefig(fn, bbox_inches='tight')
    print("Figure created: {}".format(fn))


if __name__ == '__main__':
    main()

File: targetcalib-pSCT/scripts/mapping/plot_data_tc.cc
/*
Plot data using TargetCalib and ROOT
Compile this script using the Makefile
*/
#include <sstream>
#include <fstream>
#include <iostream>
#include "TH2.h"
#include "TCanvas.h"
#include "TArrow.h"
#include "TText.h"
#include "TargetCalib/CameraConfiguration.h"

void plot(const char* fn="input_data.txt", const char* camera_version="1.0.1")
{

  auto camera_config = CTA::TargetCalib::CameraConfiguration(camera_version);
  auto m = camera_config.GetMapping();

  TH2F* h = new TH2F("h", "", 48, -0.5, 47.5, 48, -0.5, 47.5);

  std::ifstream infile(fn);
  if (!infile.is_open()) {
      std::cout << "Cannot find data file: " << fn << std::endl;
      return;
  }

  int pix = 0;
  float val;
  while (infile >> val){
    int x = m.GetColumn(pix);
    int y = m.GetRow(pix);
    h->SetBinContent(x,y,val);
    ++pix;
  }
  infile.close();

  TCanvas *c1 = new TCanvas("", "", 1200, 1200);
  h->Draw("colz");
  h->SetStats(0);

  int axl = m.fOTUpCol_l;
  int ayl = m.fOTUpRow_l;
  int axu = m.fOTUpCol_u;
  int ayu = m.fOTUpRow_u;
  TArrow* arr = new TArrow(axl, ayl, axu, ayu, 0.02);
  arr->SetLineColor(2);
  arr->SetFillColor(2);
  arr->Draw();
  TText *text = new TText();
  text -> SetTextColor(2);
  text -> SetTextSize(0.02);
  text -> SetTextAlign(21);
  text -> DrawText(axl, ayl , "ON-Telescope UP");
  c1->SaveAs("image_r_plot_data_tc.pdf");

}


int main(int argc, char** argv) {
  plot();
}